Library/Homebrew/dependency.rb

```diff
@@ -28,8 +29,49 @@ def hash
     name.hash ^ tags.hash
   end
 
+  def tap
+    CoreTap.instance
+  end
+
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def closest_spec_for_dependency_upgrade_sym
+    rack = HOMEBREW_CELLAR.join(name.split("/").last)
+    return :stable unless rack.exist?
+
+    formula = Formulary.from_rack(rack)
+    return :stable if formula.tap != tap
+
+    active_spec_sym = formula.active_spec_sym
+    return :stable if active_spec_sym == :stable
+    return formula.active_spec_sym unless formula.any_version_installed?
```

Comment:
`return active_spec_sym unless formula.any_version_installed?`? Also, not sure I understand how this would be hit if we have `return :stable unless rack.exist?` above?